src/cljs/main/broadfcui/common/notifications.cljs

```diff
@@ -107,60 +108,80 @@
-                                  :border "2px solid white" :borderRadius "0.25rem"
-                                  :padding "0.5rem 1rem" :marginLeft "0.5rem" :flexShrink 0}
-                          :href (if external? url (when-not loading? "javascript:;"))
-                          :onClick (when-not (or external? loading?)
-                                     (fn []
-                                       (utils/ajax-orch
-                                        "/profile/trial"
-                                        {:method :post
-                                         :on-done (fn []
-                                                    (profile/reload-user-profile
-                                                     #(swap! state dissoc :loading?)))})
-                                       (swap! state assoc :loading? true)))
-                          :target (when external? "_blank")}
-                      (if loading?
-                        (spinner {:style {:fontSize "1rem" :margin 0}})
-                        label)
-                      (when external?
-                        (icons/render-icon
-                         {:style {:margin "-0.5em -0.3em -0.5em 0.5em" :fontSize "1rem"}}
-                         :external-link))])))
-                [:button {:className "button-reset" :onClick #(swap! state assoc :dismissed? true)
-                          :style {:alignSelf "center" :fontSize "1.5rem" :padding "1rem"
-                                  :color "white" :cursor "pointer"}
-                          :title "Hide for now"}
-                 (icons/render-icon {} :close)])))))))
+     (let [{:keys [dismissed? loading? messages error]} @state]
+       (when-let [current-trial-state (keyword (:trialState @profile/saved-user-profile))]
+         (when (and (not dismissed?) (current-trial-state messages)) ; Disabled or mis-keyed users do not see a banner
+           (let [{:keys [title message warning? link button eula]} (messages current-trial-state)]
+             (apply ;; needed until dmohs/react deals with nested seq's
+              flex/box
+              {:style {:color "white"
+                       :background-color ((if warning? :state-warning :button-primary) style/colors)}}
+              (flex/box
+               {:style {:padding "1rem" :flexGrow 1
+                        :justifyContent "center" :alignItems "center"}}
+               [:div {:data-test-id "trial-banner-title"
+                      :style {:fontSize "1.5rem" :fontWeight 500 :textAlign "right"
+                              :borderRight "1px solid white"
+                              :paddingRight "1rem" :marginRight "1rem"
+                              :maxWidth 200 :flexShrink 0}}
+                title]
+               [:span {:style {:maxWidth 600 :lineHeight "1.5rem"}}
+                message
+                (when link
+                  (let [{:keys [label url]} link]
```

Comment:
My experimentation suggests that you can destructure in a `when-let` and it will switch off of whether or not the thing being destructured is `nil` or not. So I think you can `when-let` this (and right below here).